src/containers/dynamicarray.d

```diff
@@ -106,6 +106,14 @@ struct DynamicArray(T, bool supportGC = true)
 	/// Returns: the number of items in the array
 	size_t length() const nothrow pure @property @safe @nogc { return l; }
 
+	/**
+	 * Returns: a slice to the underlying array.
+	 *
+	 * As the memory of the array may be freed, access to this array is
+	 * highly unsafe.
+	 */
+	@property T[] ptr() @nogc { return arr; }
```

Comment:
yes, and T[] is T* and the length

I will change it to T* to make it crystal clear that danger lurks